politeiawww/cmd/piwww/editproposal.go

```diff
@@ -107,22 +121,55 @@ func (cmd *EditProposalCmd) Execute(args []string) error {
 
 		files = append(files, f)
 	}
-
 	// Setup metadata
-	if cmd.Name == "" {
-		r, err := util.Random(v1.PolicyMinProposalNameLength)
+	pm := v1.ProposalMetadata{}
+
+	// Use existing metadata flag is true
+	// fetch record and pre-fill data
+	if cmd.Usemd {
+		pdr, err := client.ProposalDetails(cmd.Args.Token,
+			&v1.ProposalsDetails{})
 		if err != nil {
 			return err
 		}
-		cmd.Name = hex.EncodeToString(r)
+		pm.Name = pdr.Proposal.Name
+		if pdr.Proposal.LinkTo != "" {
+			pm.LinkTo = pdr.Proposal.LinkTo
+		}
+		if pdr.Proposal.LinkBy != 0 {
+			pm.LinkBy = pdr.Proposal.LinkBy
+		}
 	}
-	pm := v1.ProposalMetadata{
-		Name: cmd.Name,
+	if cmd.Random {
+		// Both random & name flags used
+		// throw error
+		if cmd.Name != "" {
+			return errEditProposalRandomAndNameFound
+		}
+
+		// generate random name
+		r, err := util.Random(v1.PolicyMinProposalNameLength)
+		if err != nil {
+			return err
+		}
+		pm.Name = hex.EncodeToString(r)
 	}
+	// If RFP flag is true
+	// set linkby to a month from now
 	if cmd.RFP {
-		// Set linkby to a month from now
+		// Both rfp & linkby flags used
```

Comment:
All of the validation should be the first thing that you do in this function.  Step 1, validate all of the user provided arguments.  Step 2, run the actual code.